src/librustc_mir/interpret/validity.rs

```diff
@@ -382,14 +382,20 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
             // alignment and size determined by the layout (size will be 0,
             // alignment should take attributes into account).
             .unwrap_or_else(|| (place.layout.size, place.layout.align.abi));
+        let ptr = self.ecx.memory.check_ptr_access_align(
+            place.ptr,
+            size,
+            Some(align),
+            CheckInAllocMsg::InboundsTest,
+        );
+        if let Err(InterpErrorInfo { kind: err_unsup!(ReadBytesAsPointer), ..}) = ptr {
+            if self.ecx.tcx.features().const_int_ref {
```

Comment:
Wow what a hack. This needs *way* more comments (more than "no explanation at all", anyway. ;)

But I guess the goal here actually *is* to allow pointers that CTFE would consider dangling, so no surprise I guess.

Reply:
Oh also, feature gates shouldn't affect what we do in Miri, so please make this conditional CTFE-only.

Reply:
> please make this conditional CTFE-only.

On this note, do we have tests in Miri that make sure we still catch dangling integer references (and fn pointers)?